packages/roosterjs-content-model-plugins/lib/imageEdit/utils/getContentModelImage.ts

```diff
@@ -5,10 +5,10 @@ import type { ContentModelImage, IEditor } from 'roosterjs-content-model-types';
  * @internal
  */
 export function getContentModelImage(editor: IEditor): ContentModelImage | null {
-    const model = editor.getContentModelCopy('disconnected' /*mode*/);
+    const model = editor.getContentModelCopy('disconnected');
     const selectedSegments = getSelectedSegments(model, false /*includeFormatHolder*/);
     if (selectedSegments.length == 1 && selectedSegments[0].segmentType == 'Image') {
-        return selectedSegments[0];
+        return selectedSegments[0] as ContentModelImage;
```

Comment:
This is fine here, but if we do this kind of type cast in other place, it will be dangerous.

Actually, if you only want dataset, just get it directly from the connect model, and return the dataset instead. 

Disconnected model actually need a clone, which is not as good perf as connected one.

```ts
function getSelectedImageDataset() {
  let result;

  editor.formatContentModel(model => {
    const image = getSelectedImage(model);
    result = getImageMetadata(image);
    return false;
  });

  return result;
```

Something like this.